src/components/StaffPayment/StaffPayment.vue

```diff
@@ -172,30 +170,31 @@ export default class StaffPayment extends Vue {
 
   /** Prompt the field validations. */
   @Watch('validate')
-  private validateFields (): void {
-    this.$refs.fasForm.validate()
-    this.$refs.bcolForm.validate()
+  private validateFields (val: boolean): void {
+    if (val) {
+      if (this.paymentOption === StaffPaymentOptions.FAS) this.$refs.fasForm.validate()
+      if (this.paymentOption === StaffPaymentOptions.BCOL) this.$refs.bcolForm.validate()
```

Comment:
We only want to validate the correct form when validation is enabled. (This should also fix weirdness in the other apps that use this component.)